lib/swaps/Swaps.ts

```diff
@@ -179,41 +173,42 @@ class Swaps extends EventEmitter {
   }
 
   /**
-   * Handles a request from a peer to create a swap deal. Creates a deal,
-   * and stores the deal in the local collection of deals. Responds to the peer with a swap response packet.
+   * Checks if a payment route exists for a specified node and network.
    */
-  private handleSwapRequest = (requestPacket: packets.SwapRequestPacket, peer: Peer)  => {
-    assert(requestPacket.body, 'SwapRequestPacket does not contain a body');
-    const requestBody = requestPacket.body!;
-
-    // TODO: we always accept proposed quantity currently, put in checks to make sure we have the full amount available
-    // TODO: consider reduced quantity
-    // TODO: check that the order is still valid and take it off the order book to prevent double execution
-    // TODO: check that we have route to taker.
-    // TODO: extract data needed for proper timelock calculation and share with taker
+  private checkIfRouteExists = async (_destPubKey: string, _amount: number, _currency: string) => {
+    // TODO: implement
+    return true;
+  }
 
-    // accept the deal
+  /**
+   * Accepts a proposed deal for a specified amount and price *if and only if* a route exists to
+   * pay the taker. Stores payable deal in the local collection of deals.
+   * @returns true if a route exists to pay taker and the deal is stored, false otherwise
+   */
+  public acceptDeal = async (acceptedQuantity: number, price: number, requestBody: packets.SwapRequestPacketBody, peer: Peer) => {
+    // TODO: extract data needed for proper timelock calculation and share with taker
+    const { takerAmount, makerAmount } = Swaps.calculateSwapAmounts(acceptedQuantity, price);
     const deal: SwapDeal = {
       ...requestBody,
-      quantity: requestBody.proposedQuantity,
+      takerAmount,
+      makerAmount,
+      price,
+      quantity: acceptedQuantity,
       phase: SwapDealPhase.SwapCreated,
       state: SwapDealState.Active,
       stateReason: '',
       takerPubKey: peer.getLndPubKey(requestBody.takerCurrency),
-      r_hash: requestBody.r_hash,
       myRole: SwapDealRole.Maker,
       createTime: Date.now(),
     };
 
-    this.addDeal(deal);
-
-    const responseBody: packets.SwapResponsePacketBody = {
-      r_hash: requestBody.r_hash,
-      quantity: requestBody.proposedQuantity,
-    };
-
-    peer.sendPacket(new packets.SwapResponsePacket(responseBody, requestPacket.header.id));
-    this.setDealPhase(deal, SwapDealPhase.SwapAgreed);
+    // check that a route exists to pay our part of the swap
+    const hasRoute = await this.checkIfRouteExists(deal.takerPubKey!, deal.takerAmount, deal.takerCurrency);
```

Comment:
I would check if a route exists before creating the `deal` object because if `hasRoute` is false `deal` will be discarded

Reply:
@daniel, please make sure your changes don't conflict with #503.

Reply:
I wrote and shared these changes before #503 was opened.